src/components/typography/list.component.tsx

```diff
@@ -25,7 +31,7 @@ List.propTypes = {
   as: propTypes.oneOf(["ul", "ol"]),
 };
 
-const ListItem = ({ children, ...props }) => (
+const ListItem = ({ children, ...props }: ListProps) => (
```

Comment:
Valid `as` values will be different for `ListItem` and `List` so they should have slightly different prop interfaces

Reply:
agreed - except I don't think anything other than `li` is valid for list items (if the parent component is `ul` or `ol`) so maybe it's best just not to expose `as` in the props interface for `ListItem`